### PR TITLE
chore: add a formatter for protobufs

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- opportunistic fmt target (if clang-format is available then format)

- google style: https://developers.google.com/protocol-buffers/docs/style indent 2, col limit 80

## Commentary (optional)

suggestions on format settings? 

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: proto/.clang-format [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+BasedOnStyle: Google
+IndentWidth: 4
+---
+Language: Proto
+ColumnLimit: 100
+AlignConsecutiveAssignments: true
```
Comment (hamidzr) [inline proto/.clang-format:7]: this will result in unrelated lines (from a legitimate change) getting formatted.

Comment (hamidzr) [reply proto/.clang-format:7]: yay or nay?

File: proto/.clang-format [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+BasedOnStyle: Google
+IndentWidth: 4
```
Comment (hamidzr) [inline proto/.clang-format:3]: I personally like 2 better but 4 seems like the standard.

File: proto/.clang-format [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+---
+Language: Proto
+ColumnLimit: 100
```
Comment (stoksc) [inline proto/.clang-format:6]: I'm anti line-length so I vote to remove this entirely.

Comment (stoksc) [reply proto/.clang-format:6]: I really want to bump our go line length cap to 120.

File: proto/.clang-format (hunk 2 above)
Comment (stoksc) [reply proto/.clang-format:3]: I like 4, just because it lines up with go tabs better, but you're usually coming from ts which is 2 right?

File: proto/.clang-format (hunk 1 above)
Comment (stoksc) [reply proto/.clang-format:7]: yay, same as go

File: proto/Makefile [hunk 4]
```diff
@@ -33,3 +33,9 @@ build: $(patsubst %.proto, %.pb.go, $(shell find src/determined -type f -name '*
 .PHONY: check
 check:
 	buf check lint
+
+.PHONY: fmt
+fmt:
+	type clang-format > /dev/null || exit 0
```
Comment (stoksc) [inline proto/Makefile:39]: IMO we should add this to CI if we're going to do this.

Comment (hamidzr) [reply proto/Makefile:39]: hmm do you mean we should add the dependency? we don't really run fmt in CI. it'd be good to have `check` in CI which we don't have here 

I have this line here because I didn't want to worry about adding this dependency to our toolchain just yet and opportunistically format for now.

**Comment by hamidzr** [comment]
any idea what this is about? https://app.circleci.com/pipelines/github/determined-ai/determined/6239/workflows/35554d59-0943-42d9-a0f7-8920d82c5a32/jobs/163842

File: proto/.clang-format (hunk 1 above)
Comment (dzhu) [reply proto/.clang-format:7]: _(pet peeve: the word you want in this context is "[yea](https://grammarist.com/usage/yea-yeah-yay/)")_

File: proto/Makefile (hunk 4 above)
Comment (dzhu) [reply proto/Makefile:39]: Well, we've seen from the reST case how well it works to expect people to apply this sort of thing if they don't actually have to. :stuck_out_tongue: And there's a standard package for a reasonably recent version (8) even on Ubuntu 16.04, so it shouldn't be that big a deal to install. Moderate vote from me for going straight to checking it automatically, since we know the tool is stable.

(There's not a handy `--check`, but it looks like you can do `clang-format --dry-run --Werror` to get checking behavior.)

File: proto/.clang-format (hunk 2 above)
Comment (hamidzr) [reply proto/.clang-format:3]: > ts which is 2 right?

we picked 2, I'm not sure if there is consensus on what's standard.

Comment (hamidzr) [reply proto/.clang-format:3]: lets go with 4

File: proto/.clang-format (hunk 3 above)
Comment (hamidzr) [reply proto/.clang-format:6]: hmm I'm for having a length cap. I was between picking 80 or 100 if you prefer a higher limit stick with 100 to be consistent with our Go? :thinking:

**Review by dzhu — COMMENTED** [review]
Heads-up: #1390 is also touching this Makefile, though I think it may actually not conflict.

File: proto/Makefile (hunk 4 above)
Comment (hamidzr) [reply proto/Makefile:41]: I did add the check target with the same "don't fail if clang-format is missing". next would be adding the dependency to ci and contribution guide

**Comment by hamidzr** [comment]
I looked a little into adding clang-format to ci but it means we won't be able to use the circleci default go image any more

we could either
1. build a new image with go and clang-format and the rest of our dependencies for lining and push it to our docker hub
2. use something off of docker hub
3. modify and reuse our base ci image

it seems perfectly separatable so I'd leave those for another pr. I added a check target from what @dzhu  suggested.

File: proto/.clang-format (hunk 1 above)
Comment (stoksc) [reply proto/.clang-format:7]: (i did indeed mean yay, lol.. just a joke that will never come across on github comments)

File: proto/Makefile [hunk 5]
```diff
@@ -33,3 +33,12 @@ build: $(patsubst %.proto, %.pb.go, $(shell find src/determined -type f -name '*
 .PHONY: check
 check:
 	buf check lint
+
+.PHONY: check-fmt
+check-fmt:
```
Comment (stoksc) [inline proto/Makefile:38]: I'll never remember this target exists. Can it just run as part of `check` above?

Comment (stoksc) [reply proto/Makefile:38]: Oh wait no, because then CI would fail 🤦

**Review by stoksc — APPROVED** [review]

File: proto/Makefile (hunk 5 above)
Comment (stoksc) [reply proto/Makefile:38]: Really unfortunate.

Comment (hamidzr) [reply proto/Makefile:38]: yeah but only until we add it to CI.
